library/core/functions.general.php

```diff
@@ -3788,3 +3788,40 @@ function increaseMaxExecutionTime($maxExecutionTime) {
         return true;
     }
 }
+
+if (!function_exists('slugify')) {
+    /**
+     * Converts a string to a slug-type string.
+     *
+     * Based off Symfony's Jobeet tutorial, and found here:
+     * http://stackoverflow.com/questions/2955251/php-function-to-make-slug-url-string
+     *
+     * @param string $text The text to convert.
+     * @return string mixed|string The slugified text.
+     */
+    function slugify($text) {
+        // replace non letter or digits by -
+        $text = preg_replace('~[^\pL\d]+~u', '-', $text);
```

Comment:
Nitpicking here but I found these Regex hard to read with ~ as delimiter. I would go for / since there are no slash in these anyway.